src/utils/pureComments.ts

```diff
@@ -0,0 +1,46 @@
+// @ts-ignore
+import walk from 'acorn/dist/walk';
+import * as ESTree from 'estree';
+import { CommentDescription } from '../Module';
+
+function forEachNodeAfterComment(
+	ast: ESTree.Program,
+	commentNodes: CommentDescription[],
+	handleNode: (node: ESTree.Node) => void
+): void {
+	let commentIdx = 0;
+
+	checkCommentsBeforeNode(<any>ast);
+
+	function checkCommentsBeforeNode(node: ESTree.Node & { start: number; end: number }) {
```

Comment:
Typings were missing. Unfortunately, the outdated rollup-plugin-typescript does not properly flag errors at the moment. My IDE does, though.

Looking a little through acorn, it seems that there is no reason to pass state around so I switched to local variables to avoid the property access. By inlining the handler, there was no reason to construct an array of nodes.

Reply:
It hate/love types so much. I hate them when they cause you to have to do things like you've had to do on this line. The declaration adds no value and almost no information and looks confusing like destructuring syntax, which can already be pretty much the most complicated syntax in the language to read.

I'm not saying there's any better way, just... ugh.

Reply:
Also: given the level of scrutiny you're giving perf, the biggest perf change that could be made would be pulling this inner function definition outside of forEachNodeAfterComment. It should be possible, since the unused second parameter to this function is a state object, which could be used to store all the parameters currently captured by closing over the outer function. That way everything only needs to get hot once even if a multitude of files are checked.

Reply:
Makes sense. About the types: The big issue is that the external acorn types are wrong. They tend to work a little better and are actually quite useful when we have everything under control. Here the important information is that this accepts ESTree nodes instead of the rollup derivatives we use everywhere else. But I see you extracted it nicely